livre/core/pipeline/InputPort.cpp

```diff
@@ -0,0 +1,117 @@
+/* Copyright (c) 2011-2016, EPFL/Blue Brain Project
+ *                     Ahmet Bilgili <ahmet.bilgili@epfl.ch>
+ *
+ * This file is part of Livre <https://github.com/BlueBrain/Livre>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include <livre/core/pipeline/InputPort.h>
+#include <livre/core/pipeline/OutputPort.h>
+#include <livre/core/pipeline/Promise.h>
+#include <livre/core/pipeline/Future.h>
+
+namespace livre
+{
+
+struct InputPort::Impl
+{
+    Impl( const DataInfo& info )
+        : _info( info )
+    {}
+
+    ~Impl()
+    {}
+
+    std::string getName() const
+    {
+        return _info.first;
+    }
```

Comment:
To reduce code size you can omit this function as well as getSize and getDataType, they would still be onliners in the InputPort methods.